### PR TITLE
arm64 fix: Replaced float value strong equal checks with check with tolerance

### PR DESCRIPTION
**Pull request by asmorkalov**
Fixes: https://github.com/opencv/opencv/issues/19117

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/gapi/test/common/gapi_core_tests_inl.hpp [hunk 1]
```diff
@@ -1297,7 +1297,7 @@ TEST_P(PhaseTest, AccuracyTest)
     // Comparison //////////////////////////////////////////////////////////////
     // FIXME: use a comparison functor instead (after enabling OpenCL)
     {
-        EXPECT_EQ(0, cvtest::norm(out_mat_ocv, out_mat_gapi, NORM_INF));
+        EXPECT_LT(cvtest::norm(out_mat_ocv, out_mat_gapi, NORM_INF), 4e-6);
```
Comment (dmatveev) [inline modules/gapi/test/common/gapi_core_tests_inl.hpp:1300]: hmm in google tests' `EXPECT(a,b)` `a` is a golden reference usually, does this mean it should be `EXPECT_GT(4e-6, norm..))` ?

**Review by dmatveev — APPROVED** [review]

File: modules/gapi/test/common/gapi_core_tests_inl.hpp (hunk 1 above)
Comment (alalek) [reply modules/gapi/test/common/gapi_core_tests_inl.hpp:1300]: This rule is for `_EQ` only due to output messages (LT,LE,GT,GE doesn't require that)

Comment (alalek) [inline modules/gapi/test/common/gapi_core_tests_inl.hpp:1300]: Please put this ARM check under:
```
#if defined(__aarch64__) || defined(__arm__)
```

guard and keep original check "as is".


BTW, almost all failures on ARM64 are about the "Fluid" backend.
CPU (OpenCV) backend results should stay the same.
It make sense to G-API team implement the "FIXME" comment about the comparison functor as test parameter later.

Comment (GArik) [reply modules/gapi/test/common/gapi_core_tests_inl.hpp:1300]: Why not `EXPECT_FLOAT_EQ(0, cvtest::norm())` / `EXPECT_NEAR()` ? 4e-6 looks like a magic number to me.

File: modules/gapi/test/common/gapi_core_tests_inl.hpp [hunk 2]
```diff
@@ -1316,7 +1316,7 @@ TEST_P(SqrtTest, AccuracyTest)
...
```
Comment (GArik) [inline modules/gapi/test/common/gapi_core_tests_inl.hpp:1319]: Ditto.

**Review by alalek — APPROVED** [review]

**Pull request merged**